packages/cactus-plugin-satp-hermes/src/main/typescript/core/recovery/rollback/stage2-rollback-strategy.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger, LoggerProvider } from "@hyperledger/cactus-common";
+import { SATPSession } from "../../satp-session";
+import { RollbackState, RollbackStrategy } from "./rollback-strategy-factory";
+import { RollbackLogEntry } from "../../../generated/proto/cacti/satp/v02/crash_recovery_pb";
+import { SATPBridgeManager } from "../../stage-services/satp-bridge/satp-bridge-manager";
+
+export class Stage2RollbackStrategy implements RollbackStrategy {
+  private log: Logger;
+  private bridgeManager: SATPBridgeManager;
+  private rollbackLogEntry: RollbackLogEntry;
+
+  constructor(bridgeManager: SATPBridgeManager, logEntry: RollbackLogEntry) {
+    this.log = LoggerProvider.getOrCreate({ label: "Stage2RollbackStrategy" });
+    this.bridgeManager = bridgeManager;
+    this.rollbackLogEntry = logEntry;
+  }
+
+  async execute(session: SATPSession): Promise<RollbackState> {
+    const fnTag = "Stage2RollbackStrategy#execute";
+    this.log.info(`${fnTag} Executing rollback for Stage 2`);
+
+    if (!session) {
+      throw new Error(`${fnTag}, session data is not correctly initialized`);
+    }
+
+    try {
+      // TODO: Implement Stage 2 specific rollback logic
+
+      // TODO: Record the rollback on the log. Implement RollbackLogEntry
```

Comment:
Please un-hardcode the log creation where applicable. 
For instance, details can be a subset of the execution trace to the moment. Remember, these logs should help developers finding bugs or understanding the state of the application at any moment